pinot-broker/src/test/java/com/linkedin/pinot/broker/broker/HelixBrokerStarterTest.java

```diff
@@ -207,6 +219,40 @@ public Boolean call() throws Exception {
         .values().iterator().next().size(), SEGMENT_COUNT);
   }
 
+  @Test
+  public void testTableSegmentRefresh() throws Exception {
+    // This test verifies that when the segments of an offline table are refreshed, the TimeBoundaryInfo is also updated
+    // to a newer timestamp.
+    TimeBoundaryService.TimeBoundaryInfo tbi =
+            _helixBrokerStarter.getHelixExternalViewBasedRouting().
+                    getTimeBoundaryService().getTimeBoundaryInfoFor(DINING_TABLE_NAME);
+
+    Assert.assertEquals(tbi.getTimeValue(), "10");
+
+    List<String> segmentNames = _pinotResourceManager.getSegmentsFor(DINING_TABLE_NAME);
+    long endTime = 20L;
+    // Refresh all 5 segments.
+    for (String segment : segmentNames) {
+      OfflineSegmentZKMetadata offlineSegmentZKMetadata =
+              _pinotResourceManager.getOfflineSegmentZKMetadata(RAW_DINING_TABLE_NAME, segment);
+      Assert.assertNotNull(offlineSegmentZKMetadata);
+      _pinotResourceManager.refreshSegment(
+              SegmentMetadataMockUtils.mockSegmentMetadataWithEndTimeInfo(RAW_DINING_TABLE_NAME, segment, endTime ++),
+              offlineSegmentZKMetadata);
+    }
+    // Due to the asynchronous nature of the TimeboundaryInfo update and thread scheduling, the updated time boundary
+    // may not always be the max endtime of segments. We do not expect such exact update either as long as the timestamp
+    // is updated to some newer value.
+    waitForPredicate(() -> {
+      TimeBoundaryService.TimeBoundaryInfo timeBoundaryInfo = _helixBrokerStarter.getHelixExternalViewBasedRouting().
+              getTimeBoundaryService().getTimeBoundaryInfoFor(DINING_TABLE_NAME);
+      return 10 < Long.parseLong(timeBoundaryInfo.getTimeValue());
+    }, 30000L);
```

Comment:
Better get the sleeptimemillisecconds from the config use that here instead of hard-coding 30s and leading to unpredictability if someone changes the sleep interval.

Reply:
Done.